add bucket replicate component

yeya24 opened the pull request:
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

<!--
    Keep PR title verbose enough and add prefix telling
    about what components it touches e.g "query:" or ".*:"
-->

<!--
    Don't forget about CHANGELOG!

    Changelog entry format:
    - [#<PR-id>](<PR-URL>) Thanos <Component> ...

    <PR-id> Id of your pull request.
    <PR-URL> URL of your PR such as https://github.com/thanos-io/kube-thanos/pull/<PR-id>
    <Component> Component affected by your changes such as Query, Store, Receive.
-->

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

<!-- Enumerate changes you made -->

Add a new bucket replicate component. The libsonnet file is modified based on the bucket.libsonnet file.

## Verification

<!-- How you tested it? How do you know it works? -->

We have been using this component in production.

yeya24 commented:
cc @kakkoyun @metalmatze

kakkoyun reviewed (approved):
LGTM. I don't get why minio related manifests have changed. Did we miss something feom previous PRs?

yeya24 commented on examples/development-minio/pvc.yaml:
```diff
@@ -2,10 +2,10 @@ apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   name: minio
-  namespace: thanos
```
I think we can remove this line as other minio manifests don't have namespace specified as well.

yeya24 commented:
> LGTM. I don't get why minio related manifests have changed. Did we miss something feom previous PRs?

Thanks for catching this. I updated some minio manifests mistakenly and I already changed them back. PTAL

yeya24 commented:
Ready for review again. CC @kakkoyun
